### PR TITLE
Move to psycopg2 2.5+

### PR DESCRIPTION
**Pull request by bersace**
Still use spc for replication connection as it's only available in
psycopg2 2.7+.

### CODE REVIEW

File: temboardagent/plugins/dashboard/metrics.py [hunk 1]
```diff
@@ -203,22 +203,20 @@ def get_memory_usage(self,):
...
+            pg_database.datname))::bigint) as total_size,
+        to_char(now(),'HH24:MI') as time,
+        sum(xact_commit)::BIGINT as total_commit,
+        sum(xact_rollback)::BIGINT as total_rollback
+        FROM
+        pg_database
+        JOIN pg_stat_database ON (pg_database.oid = pg_stat_database.datid)
+        WHERE
+        datistemplate = 'f'
+        """)
```
Comment (pgiraud) [inline temboardagent/plugins/dashboard/metrics.py:219]:
```suggestion
        row = self.conn.query1("""\
        SELECT
          count(datid) as databases,
          pg_size_pretty(sum(pg_database_size(
            pg_database.datname))::bigint) as total_size,
          to_char(now(),'HH24:MI') as time,
          sum(xact_commit)::BIGINT as total_commit,
          sum(xact_rollback)::BIGINT as total_rollback
        FROM pg_database
        JOIN pg_stat_database ON (pg_database.oid = pg_stat_database.datid)
        WHERE datistemplate = 'f'
        """)
```

File: temboardagent/postgres.py [hunk 2]
```diff
@@ -2,26 +2,61 @@
...
+        with self.cursor() as cur:
+            cur.execute(dedent(query), vars)
+
+    def query(self, query, vars=None):
+        with self.cursor() as cur:
+            cur.execute(dedent(query), vars)
+            for row in cur:
+                yield row
+
+    def query1(self, query, vars=None):
```
Comment (pgiraud) [inline temboardagent/postgres.py:34]: Just a matter of taste, I would have prefered `query_one`.

File: tests/func/run_tests_docker.sh [hunk 3]
```diff
@@ -31,10 +31,16 @@ then
     # Search for the proper RPM package
     rpmdist=$(rpm --eval '%dist')
     rpm=$(readlink -e dist/rpm/noarch/temboard-agent-*${rpmdist}*.noarch.rpm)
-    yum -d1 install -y $rpm
+    # Disable pgdg to used base pyscopg2 2.5 from Red Hat.
```
Comment (pgiraud) [inline tests/func/run_tests_docker.sh:34]: to use

File: tests/func/test_activity.py [hunk 4]
```diff
@@ -23,91 +15,58 @@ def pg_sleep(duration=1):
...
-                     database='postgres')
-    try:
-        conn.connect()
-        conn.execute("CREATE DATABASE %s" % (dbname))
-        conn.close()
-    except error:
-        pass
+    with pgconnect() as conn:
+        cur = conn.cursor()
+        cur.execute("COMMIT;")
```
Comment (pgiraud) [inline tests/func/test_activity.py:31]: What is this commit supposed to do?

**Comment by pgiraud** [comment]
Is it required to use a `\` in the multiline strings?

**Comment by pgiraud** [comment]
I get the following error while trying the `maintenance` plugin:
```
ERROR: connection already closed
ERROR: Traceback (most recent call last):
ERROR:   File "/usr/local/src/temboard-agent/temboardagent/httpd.py", line 109, in response
ERROR:     (code, message) = self.route_request()
ERROR:   File "/usr/local/src/temboard-agent/temboardagent/httpd.py", line 263, in route_request
ERROR:     return (200, func(http_context, self.app))
ERROR:   File "/usr/local/src/temboard-agent/temboardagent/plugins/maintenance/__init__.py", line 25, in get_instance
ERROR:     for database in rows:
ERROR:   File "/usr/local/src/temboard-agent/temboardagent/postgres.py", line 29, in query
ERROR:     with self.cursor() as cur:
ERROR: psycopg2.InterfaceError: connection already closed
ERROR: Internal error
 INFO: client: 192.168.128.1 request: "GET /maintenance?key=key_for_agent0 HTTP/1.1" 500 -
```

**Comment by pgiraud** [comment]
I also get the following errors with the `monitoring` plugin:
```
ERROR: Unable to run probe "'port'" on "sessions" on database "main": postgres
ERROR: Unable to run probe "'port'" on "xacts" on database "main": postgres
ERROR: Unable to run probe "'port'" on "locks" on database "main": postgres
ERROR: Unable to run probe "'port'" on "blocks" on database "main": postgres
ERROR: Unable to run probe "'port'" on "bgwriter" on database "main": postgres
ERROR: Unable to run probe "'port'" on "db_size" on database "main": postgres
ERROR: Unable to run probe "'port'" on "tblspc_size" on database "main": postgres
ERROR: Unable to run probe "'port'" on "wal_files" on database "main": postgres
ERROR: Unable to run probe "'port'" on "temp_files_size_delta" on database "main": postgres
ERROR: Unable to run probe "'port'" on "heap_bloat" on database "main": postgres
ERROR: Unable to run probe "'port'" on "btree_bloat" on database "main": postgres
```

File: docker-compose.yml [hunk 5]
```diff
@@ -37,6 +37,7 @@ services:
       - run:/var/run/postgresql
     labels:
       com.dnsdock.alias: instance.tba.docker
+    command: ["-c", "shared_preload_libraries=pg_stat_statements"]
```
Comment (bersace) [inline docker-compose.yml:40]: cc @pgiraud

Comment (pgiraud) [reply docker-compose.yml:40]: I've done the following on temboardUI https://github.com/dalibo/temboard/commit/77580c2a60882266ba124088806477948fa22856

File: temboardagent/plugins/dashboard/metrics.py [hunk 6]
```diff
@@ -203,22 +203,20 @@ def get_memory_usage(self,):
...
```
Comment (pgiraud) [inline temboardagent/plugins/dashboard/metrics.py:219]: Is it possible to make it more readable? ie. avoid linebreak after FROM and WHERE, and indent fields.

Comment (pgiraud) [reply temboardagent/plugins/dashboard/metrics.py:219]: Fixed in #439

**Pull request merged**